hare/algorithm.go

```diff
@@ -213,6 +213,9 @@ type consensusProcess struct {
 	eligibilityCount uint16
 	clock            RoundClock
 	once             sync.Once
+	// layerTime is used to calculate the difference between expected message
+	// arrival times and actual message arrival times
+	layerTime time.Time
```

Comment:
expected hare start time can be always computed. 

hare_layer_time = genesis + layer*layer_duration + hare_wake_up

round_time = hare_layer_time + (round*iteration)*round_duration

layer and (round*iteration) is already encoded in the message, so time can be always computed without additional context

Reply:
Great!

Reply:
Actually, I wasn't even using it, since the round clock included everything that was required.